extensions/wikia/UserRenameTool/UserRenameToolController.class.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+class UserRenameToolController extends SpecialPage {
+	private $oldUsername;
+	private $newUsername;
+	private $reason;
+	private $token;
+	private $notifyRenamed;
+	private $confirmAction;
+	private $showConfirm = true;
+
+	private $errors = [];
+	private $warnings = [];
+	private $info = [];
+
+	public function __construct() {
+		parent::__construct( 'UserRenameTool', 'renameuser', true );
+	}
+
+	public function execute( $par ) {
+		$this->setup();
+		$this->getRequestData( $par );
+		$this->runRenameProcess();
+		$this->renderTemplate();
+
+		return;
+	}
+
+	private function setup() {
+		global $wgOut, $wgStatsDBEnabled, $wgJsMimeType, $wgUser;
+
+		$this->setHeaders();
+		$oAssetsManager = AssetsManager::getInstance();
+		$sSrc = $oAssetsManager->getOneCommonURL( '/extensions/wikia/UserRenameTool/js/UserRenameTool.js' );
+		$wgOut->addScript( "<script type=\"{$wgJsMimeType}\" src=\"{$sSrc}\"></script>" );
+
+		if ( wfReadOnly() || !$wgStatsDBEnabled ) {
+			$wgOut->readOnlyPage();
+
+			return;
+		}
+
+		if ( !$wgUser->isAllowed( 'renameuser' ) ) {
+			throw new PermissionsError( 'renameuser' );
+		}
+	}
+
+	private function getRequestData( $par ) {
+		global $wgRequest, $wgUser;
+
+		$this->oldUsername = $wgRequest->getText( 'oldusername', $par );
+		$this->newUsername = $wgRequest->getText( 'newusername' );
+		$this->reason = $wgRequest->getText( 'reason' );
+		$this->token = $wgUser->getEditToken();
+		$this->notifyRenamed = $wgRequest->getBool( 'notify_renamed', false );
+		$this->confirmAction = $wgRequest->wasPosted() && $wgRequest->getInt( 'confirmaction' );
+	}
+
+	private function runRenameProcess() {
+		global $wgRequest, $wgUser;
+
+		if ( $wgRequest->wasPosted() && $wgUser->matchEditToken( $wgRequest->getText( 'token' ) ) ) {
```

Comment:
Can use `F::app()->wg->Request->isValidWriteRequest( F::app()->wg->user )` here.  It throws an exception if its not valid so that could either be left as is or trapped if we want to send more info back.  For the same functionality as currently, would be:
```php
try {
    F::app()->wg->Request->isValidWriteRequest( F::app()->wg->user )
} catch (Exception $e) {
    return;
}
```
This has the advantage of code reuse as well as reducing the indent level of this function.